common/src/main/java/dev/ftb/mods/ftblibrary/sidebar/SidebarGroupGuiButton.java

```diff
@@ -11,152 +14,499 @@
 import net.minecraft.client.renderer.Rect2i;
 import net.minecraft.client.resources.language.I18n;
 import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
 
 import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedList;
 import java.util.List;
-
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
 
 public class SidebarGroupGuiButton extends AbstractButton {
+
 	public static Rect2i lastDrawnArea = new Rect2i(0, 0, 0, 0);
+	private static final int BUTTON_SPACING = 17;
+
+	private static final List<Component> noButtonComponents = List.of(
+			Component.translatable("sidebar_button.ftblibrary.config"),
+			Component.translatable("sidebar_button.ftblibrary.config.enter_edit_mode"));
 
-	private final List<SidebarGuiButton> buttons;
 	private SidebarGuiButton mouseOver;
+	private SidebarGuiButton selectedButton;
+	private GridLocation selectedLocation;
+	private int lastMouseClickButton = 0;
+	private boolean isEditMode;
+
+	private int currentMouseX;
+	private int currentMouseY;
+
+	private int mouseOffsetX;
+	private int mouseOffsetY;
+
+	private int currentGirdWidth = 1;
+	private int currentGridHeight = 1;
+
+	private boolean addBoxOpen;
+
+	boolean gridStartBottom = false;
+	boolean gridStartRight = false;
+
+	int yRenderStart;
+	int xRenderStart;
+
+	private boolean isMouseOverAdd;
+	private boolean mouseOverSettingsIcon;
+
+    private final Map<SidebarGuiButton, GridLocation> realLocationMap = new HashMap<>();
 
 	public SidebarGroupGuiButton() {
 		super(0, 0, 0, 0, Component.empty());
-		buttons = new ArrayList<>();
+		ensureGridAlignment();
+
 	}
 
 	@Override
 	public void renderWidget(GuiGraphics graphics, int mx, int my, float partialTicks) {
-		buttons.clear();
-		mouseOver = null;
-		int rx, ry = 0;
-		boolean addedAny;
-
-		for (var group : SidebarButtonManager.INSTANCE.getGroups()) {
-			rx = 0;
-			addedAny = false;
-
-			for (var button : group.getButtons()) {
-				if (button.isActuallyVisible()) {
-					buttons.add(new SidebarGuiButton(rx, ry, button));
-					rx++;
-					addedAny = true;
+		graphics.pose().pushPose();
+		{
```

Comment:
Not a fan of the use of `{` here. @desht thoughts?

Reply:
Going to remove this was not consistent with in the code anyway, mostly used to try and debug issues I was having